bundle/src/main/java/com/adobe/acs/commons/properties/filter/TemplatedFilter.java

```diff
@@ -0,0 +1,270 @@
+                shouldProcess = false;
+                break;
+            }
+        }
+        log.debug("URL path {} is processed: {}", urlPath, shouldProcess);
+        return shouldProcess;
+    }
+
+    /**
+     * Delegate replacement to different object types.
+     *
+     * @param node Input JSON node
+     */
+    private void replaceInElements(JsonNode node) {
+        if (node.isArray()) {
+            replaceInArray(node);
+        } else if (node.isObject()) {
+            replaceInObject(node);
+        }
+    }
+
+    /**
+     * Iterates through array items and replaces any placeholders found.
+     *
+     * @param node Array node
+     */
+    private void replaceInArray(JsonNode node) {
+        if (node.get(0) != null && node.get(0).isTextual()) {
+            List<String> updated = new LinkedList<>();
+            for (JsonNode arrayItem : node) {
+                String current = arrayItem.asText();
+                updated.add(replaceInString(current));
+            }
+            ((ArrayNode) node).removeAll();
+            for (int i = 0; i < updated.size(); i++) {
+                ((ArrayNode) node).insert(i, updated.get(i));
+            }
+        } else if (node.get(0) != null && node.get(0).isContainerNode()) {
+            for (JsonNode arrayItem : node) {
+                replaceInElements(arrayItem);
+            }
+        }
+    }
+
+    /**
+     * Iterates over keys to replace any placeholders in the values.
+     *
+     * @param node Object node
+     */
+    private void replaceInObject(JsonNode node) {
+        Iterator<String> fieldNames = node.fieldNames();
+        while (fieldNames.hasNext()) {
+            String name = fieldNames.next();
+            JsonNode nodeValue = node.get(name);
+            if (nodeValue.isContainerNode()) {
+                replaceInElements(nodeValue);
+            } else if (nodeValue.isTextual()) {
+                String current = nodeValue.asText();
+                String replaced = replaceInString(current);
+                if (!StringUtils.equals(current, replaced)) {
+                    ((ObjectNode) node).put(name, replaced);
+                }
+            }
+        }
+    }
+
+    /**
+     * Reusable method to replace placeholders in the input string./
+     *
+     * @param input String input
+     * @return The replaced or original String
+     */
+    private String replaceInString(String input) {
+        // iterate over input instead
+        while (TemplateReplacementUtil.hasPlaceholder(input)) {
+            // Get the current placeholder in the string
+            String placeholder = TemplateReplacementUtil.getPlaceholder(input);
+
+            // Transform it to the key in the property map
+            String key = TemplateReplacementUtil.getKey(placeholder);
+
+            // If the placeholder key is in the map then replace it
+            if (properties.containsKey(key)) {
```

Comment:
Could this result in an infinite loop if this `if` check is false?

Reply:
Not anymore! I updated it to use the same method that's used in the `TemplatedTransformer` - a finite loop over existing placeholders